include/deal.II/grid/grid_tools.h

```diff
@@ -1285,6 +1296,93 @@ namespace GridTools
       const bool                                             perform_handshake,
       const bool enforce_unique_mapping = false);
 
+
+    /**
+     * Data structure returned by
+     * GridTools::internal::distributed_compute_intersection_locations(). It can
+     * be converted to
+     * GridTools::internal::DistributedComputePointLocationsInternal, which can
+     * be used to reinit Utilities::MPI::RemotePointEvaluation.
+     */
+    template <int dim, int spacedim>
+    struct DistributedComputeIntersectionLocationsInternal
+    {
+      /**
+       * Intersection answers are assumed to be simplices (e.g. provided by
+       * CGAL)
+       */
+      using IntersectionAnswerType =
+        std::array<dealii::Point<spacedim>, dim + 1>;
+
+      /**
+       * Information of each intersection on sending/evaluation side. The
+       * elements of the tuple are as follows: 0) cell level and index, 1) rank
+       * of the owning process, 2) local index of the owning process, 3)
+       * found intersection, 4) requested intersection.
+       *
+       * @note The vector is sorted according to 1), 2).
+       */
+      std::vector<std::tuple<std::pair<int, int>,
+                             unsigned int,
+                             unsigned int,
+                             IntersectionAnswerType>>
+        send_components;
+
+      /**
+       * Ranks to send to.
+       */
+      std::vector<unsigned int> send_ranks;
+
+      /**
+       * Information of each received data value. The elements of the tuple are
+       * as follows: 0) rank of sender, 1) local index, 2) found intersections.
+       *
+       * @note The vector is sorted according to 1), 0), 2).
+       *
+       * @note Multiple intersections between cells can be found
+       */
+      std::vector<
+        std::tuple<unsigned int, unsigned int, IntersectionAnswerType>>
+        recv_components;
+
+      /**
+       * Pointers of ranges to found intersections for requested intersection.
+       */
+      std::vector<unsigned int> recv_ptrs;
+
+      /**
+       * Ranks from where data is received.
+       */
+      std::vector<unsigned int> recv_ranks;
```

Comment:
... and this, since the information is encoded in `send_components` and `recv_components`. One can obtain the relevant data during `convert_to_distributed_compute_point_locations_internal()` if needed.

Reply:
I would like to keep at least, ```std::vector<unsigned int> recv_ptrs``` since it is convenient to have the pointers to the found intersections in many cases. I.e. we can not only use the class to convert it to ```RPE``` but also to build parallel distributed intersection grids, etc.

Reply:
I removed recv_ranks in [a1cd90d](https://github.com/dealii/dealii/pull/15165/commits/a1cd90dfa72b54132a40dbc5b49c8d62709c98fa)